### PR TITLE
[MER-402] Show auth providers login buttons if config vars are enabled

### PR DESCRIPTION
**Pull request by gastonabella**
[MER-402](https://eliterate.atlassian.net/browse/MER-402)

### CODE REVIEW

File: test/config/auth_providers_config.exs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import Config
```
Comment (jrissler) [inline test/config/auth_providers_config.exs:1]: What about using test.exs for this vs a new file, and would avoid having to read an extra file?

**Review by jrissler — APPROVED** [review]
Looks great - left comment for you

File: lib/oli_web/pow/pow_helpers.ex [hunk 2]
```diff
@@ -229,4 +201,45 @@ defmodule OliWeb.Pow.PowHelpers do
...
+    []
+    |> maybe_add_provider(:github)
+    |> maybe_add_provider(:google)
+  end
+
+  defp maybe_add_provider(providers_list, provider) do
+    client_id = Application.fetch_env!(:oli, :auth_providers)[:"#{provider}_client_id"]
+    client_secret = Application.fetch_env!(:oli, :auth_providers)[:"#{provider}_client_secret"]
+
+    if client_id != "" and client_secret != "" do
```
Comment (mcass19) [inline lib/oli_web/pow/pow_helpers.ex]: This could also be nil right? (if they keys are not configured)

**Review by mcass19 — COMMENTED** [review]
Great work!!! Just left a minor question

File: lib/oli_web/pow/pow_helpers.ex (hunk 2 above)
Comment (gastonabella) [reply lib/oli_web/pow/pow_helpers.ex]: Actually with the current code and config it shouldn't happen, but might cause some confusion in the future. Updated! Thanks 🙌

File: test/config/auth_providers_config.exs (hunk 1 above)
Comment (gastonabella) [reply test/config/auth_providers_config.exs:1]: I can set this dummy config in `test.exs` to test the happy path, but in order to test the case when any of these env vars are missing I would need to read another config in a separate file.
Another option could be to update the env vars inside the test case (using `Application.put_env/3`) and ensure that on exit the original values are reset. Wdyt?

Comment (jrissler) [reply test/config/auth_providers_config.exs:1]: I think that might be a better way / more clearly communicate what your doing - wdyt? If we got that route - just double check on test failure, it doesn't remain and potentially cause funky issues in other tests.

Comment (gastonabella) [reply test/config/auth_providers_config.exs:1]: I checked and works fine for both successful and failure scenarios. Just pushed the updates, thanks!

Comment (jrissler) [reply test/config/auth_providers_config.exs:1]: Awesome nice improvement!

**Pull request closed without merging**